zbus/src/match_rule_builder.rs

```diff
@@ -200,12 +200,73 @@ impl<'m> MatchRuleBuilder<'m> {
     }
 
     /// Set 0th argument's namespace.
+    ///
+    /// This function is deprecated because the choice of `InterfaceName` was too restrictive.
+    #[deprecated = "use arg0ns instead"]
```

Comment:
nit: Ideally, deprecation should be a separate commit (follow-up). That way if we decide for any reason to revert the deprecation, it'd be a simple `git revert` call.